snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/SnapshotChecksum.java

```diff
@@ -37,14 +36,26 @@ public static SfvChecksum read(final Path checksumPath) throws IOException {
   }
 
   public static SfvChecksum calculate(final Path snapshotDirectory) throws IOException {
-    try (final var fileStream = Files.list(snapshotDirectory).sorted()) {
-      final SfvChecksum sfvChecksum =
-          createCombinedChecksum(fileStream.collect(Collectors.toList()));
-      sfvChecksum.setSnapshotDirectoryComment(snapshotDirectory.toString());
+    try (final var fileStream =
+        Files.list(snapshotDirectory).filter(SnapshotChecksum::isNotMetadataFile).sorted()) {
+      final SfvChecksum sfvChecksum = createCombinedChecksum(fileStream.toList());
+
+      // While persisting transient snapshot, the checksum of metadata file is added at the end.
+      // Hence when we recalculate the checksum, we must follow the same order. Otherwise base on
+      // the file name, the sorted file list will have a differnt order and thus result in a
+      // different checksum.
+      final var metadataFile = snapshotDirectory.resolve(FileBasedSnapshotStore.METADATA_FILE_NAME);
```

Comment:
:question: We already require that the metadata comes last in the sorted file stream, so I don't really understand why we have to handle this here.

Reply:
Ah, nevermind, this is so that we can change the name in the future, right?

Reply:
But even then, an old version would still have the old `METADATA_FILE_NAME` and so it wouldn't recognize the new file. That means we would have to keep ensuring that the file comes last in the sorted stream.

Reply:
This is to ensure that even if we change the file name, checksum calculation is correct. If we don't do this, then we can only use a filename that respects the sort order.

Of course, we cannot change the name just like that because the old versions and new version will have different file names.  If we want to change the name, it should be done in a backward compatible way. For example - one version that can recognize both filenames but still writes using old filename. And then the next version, which writes in new filename. Something like that.

Reply:
Alright, we are on the same page then :+1: